includes/API_Credentials/API_Credentials_Manager.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * Class WordPress\AI_Client\API_Credentials\API_Credentials_Manager
+ *
+ * @since n.e.x.t
+ * @package wp-ai-client
+ */
+
+namespace WordPress\AI_Client\API_Credentials;
+
+use WordPress\AiClient\AiClient;
+use WordPress\AiClient\Providers\DTO\ProviderMetadata;
+use WordPress\AiClient\Providers\Http\DTO\ApiKeyRequestAuthentication;
+
+/**
+ * Class for managing the AI API credentials for the various providers.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-import-type ProviderMetadataArrayShape from ProviderMetadata
+ */
+class API_Credentials_Manager {
+
+	private const OPTION_GROUP                = 'wp-ai-client-settings';
+	private const OPTION_PROVIDER_CREDENTIALS = 'wp_ai_client_provider_credentials';
+
+	/**
+	 * Initializes the API credentials manager.
+	 *
+	 * This method needs to be called by the consumer of this package, on the WordPress 'init' action hook.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function initialize(): void {
+		$this->collect_providers();
+		$this->register_settings();
+		$this->pass_credentials_to_client();
+
+		add_action(
+			'admin_menu',
+			function () {
+				$this->add_admin_screen();
+			}
+		);
+	}
+
+	/**
+	 * Collects metadata for all registered providers in the PHP AI Client SDK.
+	 *
+	 * Since the PHP AI Client SDK as well as the WordPress AI Client package can be loaded multiple times,
+	 * including with different namespace or class name prefixes, this method ensures that the provider metadata is
+	 * collected only once across all instances of the package.
+	 *
+	 * This unified collection mechanism allows the WordPress AI Client package to expose a single settings screen in
+	 * WordPress for managing API credentials for all providers, regardless of how many times the package is loaded and
+	 * regardless of whether a provider is only registered in one of the instances.
+	 *
+	 * To safely do that, the method uses a global variable. It stores the provider metadata array keyed by the
+	 * provider ID, and for each provider metadata it also stores a map of the AiClient class names where the provider
+	 * is registered in.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function collect_providers(): void {
+		/**
+		 * The internal global, to collect providers metadata across duplicate clients, including prefixed versions.
+		 *
+		 * @var ?array<string, ProviderMetadataArrayShape> $wp_ai_client_providers_metadata
```

Comment:
Nit: I prefer a union null when possible as it's clearer
```suggestion
		 * @var array<string, ProviderMetadataArrayShape>|null $wp_ai_client_providers_metadata
```

Reply:
Fixed